### ISSUE TITLE
Exposing base classes to make source code available in gammapy doc?

### ISSUE THREAD

**Issue by MRegeard** [gammapy/gammapy#4299]
**Gammapy version**
1.0 and dev

**Bug description**
I wanted to get the value of the containment radius of the PSF for a given energy. So I explored the doc of gammapy to find the right methods to do that. I found that `containment_radius`gives me exactly what I need. However, since `containment_radius` is coded in `gammapy.irf.psf.core.PSF`I cannot access the source code from the gammapy doc.
Of course in my case I went to the GitHub rep to see it. But I think it can be an issue because some of these useful methods are only coded there. 
Moreover, for this method, the handling of `**kwargs`is not really explicit, which can make it difficult for the user to know what can be passed through. I think that this method is not the only one in this case.

I also suggest that we put an example for what I wanted to do in the `containment_radius` methods since it would seems really common to want to get the value of the PSF at given coordinates.

**Comment by adonath** [comment]
Thanks @MRegeard! As far as I can tell this is a known issue in Sphinx https://github.com/sphinx-doc/sphinx/issues/2196 , so it would require an upstream fix...

**Comment by MRegeard** [comment]
From what I understand, since `gammapy.irf.psf.core.PSF` is not a public class it is not shown in the docs, which means that even if the issue you mention is fixed, the source code will still be unavailable. Am I right?

**Comment by adonath** [comment]
Yes, we could make the `PSF` base class public too...but it is not meant to be used on it's own, which might create some confusion as well...

**Comment by MRegeard** [comment]
I agree that it should not be public. 
But then this is kind of a dead end. On one hand, we would like to have access to the source code of the methods because they are used in public class. On the other hand, we cannot provide the source code in the gammapy doc because it is not public.

**Comment by registerrier** [comment]
Additional code has been made publicly available in the docs. See #5852 and #5851 

A side effect of this is that it potentially forces us to follow deprecation scheme for modification of very low level API. So we need to understand how to draw the line and let people know which part of the API is considered user-level API.

**Comment by AtreyeeS** [comment]
What is the status here? I think most of the API we expose to users is public now, no? Can we close this @Astro-Kirsty ?

**Comment by Astro-Kirsty** [comment]
No this is not fixed. 

For the example Maxime gives you can see [here](https://docs.gammapy.org/dev/api/gammapy.irf.ParametricPSF.html#gammapy.irf.ParametricPSF.containment_radius) is the definition of `containment_radius`. But it is not clear what the `**kwargs` should be? `Other coordinates.`, what does  that mean..
